Frameworks/Foundation/NSPointerArray.mm

```diff
@@ -14,146 +14,380 @@
 //
 //******************************************************************************
 
-#import <StubReturn.h>
+#import <Foundation/NSCoder.h>
+#import <Foundation/NSMutableArray.h>
 #import <Foundation/NSPointerArray.h>
+#import <Foundation/NSException.h>
+
+static NSUInteger sizeFunction(const void* item) {
+    return static_cast<NSUInteger>(sizeof(void*));
+}
+
+@implementation NSPointerArray {
+    NSUInteger _allocated;
+    NSUInteger _count;
+    void** _container;
+};
+
+- (instancetype)init {
+    if (self = [super init]) {
+        _allocated = 0;
+        _count = 0;
+        _container = nullptr;
+        _pointerFunctions = nil;
+    }
+    return self;
+}
+
+- (void)dealloc {
```

Comment:
It will be nice to override allocWithZone to prevent this class from being sub classed